### ISSUE TITLE
Admin Loads and Saves Bundle Product Child Items Price if Set to $0 - Incorrectly Inflates Product Price

### ISSUE THREAD

**Issue by PromInc** [magento/magento2#32721]
### Summary of Issue
When a child item of a bundle product has a price of `$0.00`, editing the parent bundle product in the admin sets the child items as their product price.  Then when you save the parent bundle product, those child product prices are saved and overwrites the `$0.00` set to them.  This results in an undesired inflation of the parent bundle product price.

To avoid this you have to manually set the child item price to `$0.00` prior to saving.

This issue did not exist in v2.4.1.  It was added by a code addition in v2.4.2.

### Preconditions (*)

1. Magento version 2.4.2 (not present in version 2.4.1)
2. Magento Admin

### Steps to reproduce (*)

1. Create a bundle product
2. Set at least one **Bundle Item** to the product using the **Add Option** button with the following configuration:
    1. **Input Type**: Checkbox
    1. **Required**: Yes
    1. **Is Default**: Yes
    1. **Price**: $0.00
    1. **Price Type**: Fixed
    1. **Default Quantity**: 1
3. Save product
    1. Note that the price of that child item is not set to $0.00 like was set in step 2
    1. Query this product in the database in table `catalog_product_bundle_selection` and notice that the `selection_price_value` column is set to `0.0000`
4. Save the product
    1. Query this product in the database in table `catalog_product_bundle_selection` and notice that the `selection_price_value` column is NOT set to `0.0000` - instead it's set to the price of the child product

### Expected result (*)

1. A price of `$0.00` should be respected and not overridden by the child products price

### Actual result (*)

1. The product price is saved - overriding the `$0.00` and thus un-intentionally inflating the product price.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by PromInc** [comment]
The bug is here:

```
# vendor/magento/module-bundle/Model/Product/LinksList.php
# Line 56

            $bundledProductPrice = $selection->getSelectionPriceValue();
            if ($bundledProductPrice <= 0) {
                $bundledProductPrice = $selection->getPrice();
            }
```
The comparator should be just `<` - not `<=`.

So the code should be:

```
            $bundledProductPrice = $selection->getSelectionPriceValue();
            if ($bundledProductPrice < 0) {
                $bundledProductPrice = $selection->getPrice();
            }
```

**Comment by krishprakash** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @krishprakash 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-41873` were created

**Issue Available**: @krishprakash, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by PromInc** [comment]
@magento give me 2.4-develop instance

**Comment by PromInc** [comment]
This issue can be closed as it has already been addressed in the 2.4-develop branch via commit acdefe469fbd242d66063a35f87aca202e884454

I have confirmed via the Magento instance the bug is resolved.

**Comment by magento-engcom-team** [comment]
Hi @PromInc. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-41873` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.

**Comment by iphigenie** [comment]
I am amazed this was called S3 as anyone with bundles had their bundles suddenly shown on site at inflated price that didnt show in admin.  This would result in no sales and loss of confidence in customers.

I found this and searched for it in bugs but obviously since it is marked fixed it didnt show in default search.

Suggestion: Issues that exist in the current release but which will be fixed in the next release are invisible as there is no way to get a list of them - perhaps tags "exists in" and "fixed in"?

**Comment by PromInc** [comment]
@iphigenie I agree that I don't like the labeling of this.

This bug was also reported here: https://github.com/magento/magento2/issues/32383 before me and hence I closed this ticket personally.

This bug fix is in process to be resolved in v2.4.3 via that alternate ticket.

But since the current production code is still flawed, it's strange that the other ticket is marked as fixed.  It's not live, thus the bug is still present.  Just my 2 cents.

---

And I agree - this is a nasty bug that was impacting our sales as well.  Who would buy something at 3 times the price?!?!  Trust me - our competitors are not listing their products at that price!

**Comment by iphigenie** [comment]
could #31739 come from the same set of mistakes but in LinkManagement not LinksList ?

**Comment by iphigenie** [comment]
> @iphigenie I agree that I don't like the labeling of this.
> 
> This bug was also reported here: #32383 before me and hence I closed this ticket personally.
> 
> This bug fix is in process to be resolved in v2.4.3 via that alternate ticket.
> 
> But since the current production code is still flawed, it's strange that the other ticket is marked as fixed. It's not live, thus the bug is still present. Just my 2 cents.
> 
> And I agree - this is a nasty bug that was impacting our sales as well. 

Agreed - I felt like a right idiot when I spotted it, how could I have missed it. And my bundles are all FIXED PRICE too so no matter what this should not have happened...

PS: The fact that there is only ONE bundle product and ONE grouped product in the test data, instead of a set of them using the different configuration and visibility possibility, means these bugs keep happening on these product types. Clearly the number of products and the test suite around these are insufficient. It is hard for people like me to jump in the deep end and refactor, but I'd be able, and I'd give some time to create better test data and better tests if that means these get fixed!

PPS: There really needs to be a way to see known and confirmed bugs for the current release and it also need to perhaps have a link in release notes somewhere as who would expect bundle pricing to change in undocumented ways? Closed should be released, not just "fix committed somewhere in git"

**Comment by sheepfy** [comment]
> could #31739 come from the same set of mistakes but in LinkManagement not LinksList ?

Hi, so according to 2.4-develop branch, the issue is still there.
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Bundle/Model/LinkManagement.php
